src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -61,6 +61,155 @@ static void Main()
         }
 
         [Fact]
+        public void InAttributeParameter()
+        {
+            string source = @"
+[Custom(z: default, y: default, x: default)]
+class C
+{
+    [Custom(default, default)]
+    void M()
+    {
+    }
+}
+public class CustomAttribute : System.Attribute
+{
+    public CustomAttribute(int x, string y, byte z = 0) { }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular7_1);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void InStringInterpolation()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        System.Console.Write($""({default})"");
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular7_1, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(); // Should give an error. Follow-up issue: https://github.com/dotnet/roslyn/issues/18609
+            CompileAndVerify(comp, expectedOutput: "()");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+
+            var def = nodes.OfType<LiteralExpressionSyntax>().Single();
+            Assert.Equal("default", def.ToString());
+            Assert.Null(model.GetTypeInfo(def).Type);
+            Assert.Null(model.GetTypeInfo(def).ConvertedType);
+            Assert.Null(model.GetSymbolInfo(def).Symbol);
+            Assert.False(model.GetConstantValue(def).HasValue);
+            Assert.False(model.GetConversion(def).IsNullLiteral);
+        }
+
+        [Fact]
+        public void InUsing()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        using (default)
+        {
+            System.Console.Write(""ok"");
+        }
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular7_1, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(); // Should give an error. Follow-up issue: https://github.com/dotnet/roslyn/issues/18609
```

Comment:
`using (null) { }` and `using (default(System.IDisposable)) { }` do not generate errors.